### ISSUE TITLE
Link to DF Database does only work in itself, not from a different DynamicField

### ISSUE THREAD

**Issue by StefanAbel-OTOBO** [RotherOSS/otobo#2171]
When referring to another DF in a Link, it doesn't work if that field is a DynamicField Database.
It does though, in case it calls its own value.

In the link-portion of a DF configuration, you can refer a different field value, in case you use [% Data.DynamicField_SomeFieldName | uri %]
When you have a DF DB "**DBFieldWithRefToIDinAText**" and Link to:
https://myfqdn/otobo/index.pl?Action=AdminService;Subaction=ServiceEdit;ServiceID=[% Data.DynamicField_**DBFieldWithRefToIDinAText** | uri %]

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/87359203/218459184-2ba13a59-ebda-4523-ab20-5e42f2734613.png">

this works fine:
<img width="475" alt="image" src="https://user-images.githubusercontent.com/87359203/218457667-c4d67876-3879-4105-971a-45194f119393.png">

You can see the ID being taken from the field.

When you do that in a different DynamicField called "**MyIDinAText**", with the same Link:
https://myfqdn/otobo/index.pl?Action=AdminService;Subaction=ServiceEdit;ServiceID=[% Data.DynamicField_**DBFieldWithRefToIDinAText** | uri %]
<img width="885" alt="image" src="https://user-images.githubusercontent.com/87359203/218458864-a4f14770-0ef5-4eac-9fc2-9f1c948abf6c.png">
this does not work, as hovering shows an array issue:
<img width="611" alt="image" src="https://user-images.githubusercontent.com/87359203/218457978-f11218fc-e7fb-43e1-82b7-2928b3fed355.png">

In case I would refer to a different kind field (e.g. text) this would work nicely.
As in the ticket history, the proper value (whatever you select as an Identifier for the DF DB) is stored and self-calling it in the same DF works, in my opinion this should also work from a different Dynamic Field.

**Comment by StefanAbel-OTOBO** [comment]
ov147

**Comment by stefanhaerter** [comment]
The cause of this bug is a bit complicated, I will try to explain it.

**TL;DR:** When a field links to itself, the value handling is different, which causes a problem for fields holding values as an array.

The placeholder `[% Data.DynamicField_DBField %]` is filled with the value of the field which is present when rendering the template. Usually, this is the value in the `%Ticket` object, which is an array in case of e.g. Dynamic Field Database. When linking to itself, a special case appears: Here, the function `DisplayValueRender()` is called, which returns, among other data, the Dynamic Field value as a string and overrules the Dynamic Field value stored in `%Ticket`.

I see mainly two possible solutions, assuming that we do want to support linking to other dynamic fields:

- Add values like `DynamicField_DB_DisplayValue` to `%Ticket`, holding the result of `DisplayValueRender()`
- Check if a link of a dynamic field configuration points to a dynamic field value and if so, calling `DisplayValueRender()`

@svenoe What do you think about this?

**Comment by stefanhaerter** [comment]
Discussed this with @svenoe.

The proposed solution is to access array values with indices in the template toolkit syntax, e.g. like `https://example.test/id=[% Data.DynamicField_DB.0 %]`.

Closing the issue.

Leaving the milestone as-is to indicate that the discussion affected branches from 11.0 upwards.
